middleware.go

```diff
@@ -28,3 +31,43 @@ func (r *Router) Use(mwf ...MiddlewareFunc) {
 func (r *Router) useInterface(mw middleware) {
 	r.middlewares = append(r.middlewares, mw)
 }
+
+// MethodMiddleware returns a MiddlewareFunc intended for setting the CORS
+// Method header, Access-Control-Allow-Methods. It sets Access-Control-Allow-Methods
+// on a  request, by matching routes based only on paths. It also handles
+// OPTIONS requests, by settings Access-Control-Allow-Methods, and then
+// returning without calling the next http handler.
+func MethodMiddleware(r *Router) MiddlewareFunc {
```

Comment:
Two parts:

* I think the function name could be more descriptive here: `CORSMethodMiddleware` - something to indicate that it applies to a CORS-driven use-case.
* Do we want to build the routes on every request? (may not be avoidable to avoid stale data, but thinking out loud here)

Reply:
I agree with the first part. As for building the routes on every request, I think it would be fairly difficult to do, but I don't know the mux architecture super well either. I can't think of any clean solution that keeps our data fresh. It would be nice to not have to walk the router on each request though of course, so I'm open to any ideas!